pkg/hub/custommetrics/custommetrics.go

```diff
@@ -0,0 +1,198 @@
+package custommetrics
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"html"
+	"net/http"
+	"time"
+
+	ocinfrav1 "github.com/openshift/api/config/v1"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+)
+
+var (
+	//https://..../apis/cluster.open-cluster-management.io/v1/managedclusters
+	//correspond to
+	//.../apis/G/V/R
+	mcGVR = schema.GroupVersionResource{
+		Group:    "cluster.open-cluster-management.io",
+		Version:  "v1",
+		Resource: "managedclusters",
+	}
+
+	cvGVR = schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: "clusterversions",
+	}
+
+	hubID = ""
+)
+
+//cluster_id = OCP ID of the Cluster (need to resolve for eks, etc)
+//type = K8s Distribution, e.g. OCP, EKS, etc
+//version = Distribution version
+//cluster_infrastructure_provider = value "Type" from cluster_infrastructure_provider
+//hub_id = cluster_id of hub server
+//cluster_name =User Display Name of Cluster (defaults to Id if not provided)
+var managedClusterMetric = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	Name: "a_managed_cluster",
+	Help: "Managed Cluster being managed by ACM Hub.",
+}, []string{"cluster_name", "cluster_id", "type", "version", "cluster_infrastructure_provider", "hub_id"})
+
+func getDynClient() (dynamic.Interface, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return dynamic.NewForConfig(config)
+}
+
+func getStaticData(c dynamic.Interface) {
+
+	cvObj, errCv := c.Resource(cvGVR).Get(context.TODO(), "version", metav1.GetOptions{})
+	if errCv != nil {
+		klog.Fatalf("Error getting cluster version %v \n", errCv)
+		panic(errCv.Error())
+	}
+	cv := &ocinfrav1.ClusterVersion{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(cvObj.UnstructuredContent(), &cv)
+	if err != nil {
+		klog.Fatalf("Error unmarshal cluster version object%v \n", err)
+		panic(errCv.Error())
+	}
+	hubID = string(cv.Spec.ClusterID)
+	klog.Infof("Hub id of this ACM Server is " + hubID)
```

Comment:
don't log from utility libraries.  Especially if you're going to report it later.

Reply:
agreed - removed